all_Commands/git_Commands/git.md

```diff
@@ -0,0 +1,59 @@
+# All Necessary of GIT commands
+
+Before starting any git command start with 
+
+`git config –global user.name “[name]”` and `git config –global user.email “[email address]” ` to add user to any commit
+
+## 1. git init
+`git init` help us to initialise a git in our project where we get a new ``.git`` directory in our root project folder
+
+## 2. git clone
+`git clone` help us download a repository from an existing URL
+
+## 3. git status
+`git status` gives  information on working tree status
+
+## 4. git log
+`git log` list all the version of the opened branche
+
+
+## 5. git branch
+`git banch` list all branches 
+
+
+## 6. git checkout -b <branch name>
+`git checkout -b <branch name>` help us creat a new branch
+
+
+## 7. git checkout <branch name>
+`git checkout <branch name>` help us go from one branch to another
+
+
+## 8. git add . or git add <file name>
+`git add . or git add <file name>` help us add all the changes made in the repository
+
+## 9. git commit -m"Commit message"
+`git commit -m"Commit message"` help us save and state any modification made on the repository
+
+
+## 10. git commit --amend --no-edit
+`git commit --amend --no-edit` help us save small changes made without having to state them 
```

Comment:
Not only small but also, help ue by adding changes which should be similar to the previous commit